Expand shop container ID detection with more keywords

TheHaliax opened the pull request:
The getShopContainerIds method now detects container IDs with field names containing 'shop', 'store', 'merchant', 'bazaar', 'stall', or 'trader'. This broadens support for identifying shop-related inventories by matching additional relevant keywords.

TheHaliax committed:
Expand shop container ID detection with more keywords

The getShopContainerIds method now detects container IDs with field names containing 'shop', 'store', 'merchant', 'bazaar', 'stall', or 'trader'. This broadens support for identifying shop-related inventories by matching additional relevant keywords.

gmason0 commented on runelite-client/src/main/java/net/runelite/client/plugins/microbot/MicrobotPlugin.java:
```diff
@@ -201,32 +201,43 @@ else if (Arrays.stream(getShopContainerIds()).anyMatch(sid -> Objects.equals(eve
+			String fieldName = field.getName().toLowerCase();
 
-			if (field.getName().toLowerCase().contains("shop"))
+			for (String keyword : keywords)
```
Instead of another inner for loop, we should be able to use `Arrays.stream(keywords).anyMatch(k -> fieldName.contains(k))`

TheHaliax committed:
Add 'supplies' to shop container keywords

Expanded the keywords array to include 'supplies' for identifying shop-related inventory containers. This improves detection of relevant containers in the plugin.

TheHaliax committed:
Refactor shop container ID retrieval logic

Simplified the logic for matching keywords in field names by using a stream and anyMatch. Updated Javadoc to clarify included keywords and improve description.
